modules/templates/templates_api.rb

```
@@ -0,0 +1,21 @@
+require 'templates/handler'
+class Proxy::TemplatesApi < Sinatra::Base
+  helpers ::Proxy::Helpers
+  authorize_with_trusted_hosts
+
+  get "/templateServer" do
+    begin
+      content_type :json
+      {"templateServer" => (Proxy::Templates::Plugin.settings.template_url || "")}.to_json
+    rescue => e
+      log_halt 400, e
+    end
+  end
+
+  get "/:kind" do |kind|
+    log_halt 403, "No URI specified for :foreman_url:" unless Proxy::Templates::Plugin.settings.foreman_url
```

I don't think we formally agreed on this, but I think (and I tried to follow this logic in other parts of smart-proxy) it's better to fail-fast: if a required piece of configuration is missing, the module shouldn't be loaded at all, and an appropriate message should be logged.